src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -83,12 +87,46 @@ public class PPLTool implements Tool {
 
     private String contextPrompt;
 
+    private PPLModelType pplModelType;
+
     private static Gson gson = new Gson();
 
-    public PPLTool(Client client, String modelId, String contextPrompt) {
+    private static Map<String, String> defaultPromptDict;
```

Comment:
Is there other tools would use default prompt? If so, it's better to create an abstract method to reduce the code duplication in other tools.

Reply:
I think PPL would be the current only one.

Reply:
Find Index routing tool would be another one. So add the abstractPromptedTool. Please check it.